cairo/ethereum/cancun/vm/interpreter.cairo

```diff
@@ -334,3 +353,160 @@ func _execute_code{
     // Recursive call to continue execution
     return _execute_code(evm);
 }
+
+func process_message_call{
+    range_check_ptr,
+    bitwise_ptr: BitwiseBuiltin*,
+    keccak_ptr: KeccakBuiltin*,
+    poseidon_ptr: PoseidonBuiltin*,
+}(message: Message, env: Environment) -> (MessageCallOutput, EthereumException*) {
+    alloc_locals;
+
+    // Check if this is a contract creation (target is empty)
+    if (cast(message.value.target.value.address, felt) == 0) {
```

Comment:
Bytes0(b"")  != Bytes0(b"\x00")

in this case I think it should be 

if (cast(message.value.target.value == 0) 

wdyt?

Reply:
the problem with this is that in python we have type checks, which we do not have in cairo, so this is the same as if it was address 0 (bytes20)

Reply:
I am not sure we can do that:  the target is an enum.
It would be maybe if (cast(message.value.target.value.bytes0, felt) == 0)

Reply:
The target is of type To
```
struct ToStruct {
    bytes0: Bytes0*,
    address: Address*,
}

struct To {
    value: ToStruct*,
}
```
message.value.target.value.address.value == 0 means that the pointer to Address is null hence it is empty which would be different from [message.value.target.value.address.value] which is of Address type and could be the address 0 (bytes20)

Unless I miss something?

Reply:
ah ok my bad

Reply:
its fine